src/data_readers/data_reader_image.cpp

```diff
@@ -136,6 +136,17 @@ void image_data_reader::load() {
   select_subset_of_data();
 }
 
+void image_data_reader::setup() {
+  generic_data_reader::setup();
+
+  using InputBuf_T = lbann::cv_image_type<uint8_t>;
+  auto cvMat = cv::Mat(1, get_linearized_data_size(), InputBuf_T::T(1));
+  m_thread_cv_buffer.resize(omp_get_max_threads());
+  for(int tid = 0; tid < omp_get_max_threads(); ++tid) {
+    m_thread_cv_buffer[tid] = cvMat.clone();
+  }
+}
+
```

Comment:
Nikoli, I believe that this addresses your concern.  One question for you or Jae-Seung is if any allocation from the clone is properly cleaned up when the vector is destroyed.  I believe that it should.